File: zodiak.py
from datetime import datetime

# Input dari user
tanggal = int(input("Masukan tanggal lahir: "))
bulan = int(input("Masukan bulan lahir (1-12): "))
tahun = int(input("Masukan tahun lahir: "))

# Hitung usia
today = datetime.now()
birth_date = datetime(tahun, bulan, tanggal)
age = today.year - birth_date.year - ((today.month, today.day) < (birth_date.month, birth_date.day))

# Cek usia
if age < 17:
    print("Maaf, Anda belum cukup umur (di bawah 17 tahun).")
else:
    # Tentukan zodiak
    if (bulan == 12 and tanggal >= 22) or (bulan == 1 and tanggal <= 19):
        zodiak = "Capricorn"
    elif (bulan == 1 and tanggal >= 20) or (bulan == 2 and tanggal <= 18):
        zodiak = "Aquarius"
    elif (bulan == 2 and tanggal >= 19) or (bulan == 3 and tanggal <= 20):
        zodiak = "Pisces"
    elif (bulan == 3 and tanggal >= 21) or (bulan == 4 and tanggal <= 19):
        zodiak = "Aries"
    elif (bulan == 4 and tanggal >= 20) or (bulan == 5 and tanggal <= 20):
        zodiak = "Taurus"
    elif (bulan == 5 and tanggal >= 21) or (bulan == 6 and tanggal <= 20):
        zodiak = "Gemini"
    elif (bulan == 6 and tanggal >= 21) or (bulan == 7 and tanggal <= 22):
        zodiak = "Cancer"
    elif (bulan == 7 and tanggal >= 23) or (bulan == 8 and tanggal <= 22):
        zodiak = "Leo"
    elif (bulan == 8 and tanggal >= 23) or (bulan == 9 and tanggal <= 22):
        zodiak = "Virgo"
    elif (bulan == 9 and tanggal >= 23) or (bulan == 10 and tanggal <= 22):
        zodiak = "Libra"
    elif (bulan == 10 and tanggal >= 23) or (bulan == 11 and tanggal <= 21):
        zodiak = "Scorpio"
    else:
        zodiak = "Sagitarius"  # Untuk tanggal 22 November - 21 Desember
    
    print(f"Zodiak Anda adalah: {zodiak}")
    print(f"Usia Anda: {age} tahun")

File: zodiak.cpp
/*
program menentukan zodiak
I.S : program meminta user memasukan tanggal, bulan, dan tahun lahir
F.S : program mencetak zodiak user jika usia user di atas atau sama dengan 17 tahun
F.S : program mencetak "anda masih di bawah umur" jika usia user masih di bawah 17 tahun
*/
#include <iostream>
using namespace std;

//kamus data

    int tanggal_lahir;
    int bulan_lahir;
    int tahun_lahir;
    char zodiak;

//algoritma utama
int main()
{
    int tanggal_lahir;
    cout<<"masukan tanggal_lahir:";
    cin>>tanggal_lahir;
    int bulan_lahir;
    cout<<"masukan bulan_lahir:";
    cin>>bulan_lahir;
    int tahun_lahir;
    cout<<"masukan tahun_lahir:";
    cin>>tahun_lahir;
    
if (bulan_lahir >= 9 && tahun_lahir > 2008) {
    cout<<"anda belum cukup umur";
}
else {
    
    if ((bulan_lahir == 12 && tanggal_lahir >= 22) || (bulan_lahir == 1 && tanggal_lahir <= 19)) {
        cout<<"zodiak = Capricorn";
    }
    else if ((bulan_lahir == 1 && tanggal_lahir >= 20) || (bulan_lahir == 2 && tanggal_lahir <= 18)) {
            cout<<"zodiak = Aquarius";
    }
    else if ((bulan_lahir == 2 && tanggal_lahir >= 19) || (bulan_lahir == 3 && tanggal_lahir <= 20)){
            cout<<"zodiak = Pisces";
    }
    else if ((bulan_lahir == 3 && tanggal_lahir >= 21) || (bulan_lahir == 4 && tanggal_lahir <= 19)) {
            cout<<"zodiak = Aries";
    }
    else if ((bulan_lahir == 4 && tanggal_lahir >= 20) || (bulan_lahir == 5 && tanggal_lahir <= 20)) {
            cout<<"zodiak = Taurus";
    }
    else if ((bulan_lahir == 5 && tanggal_lahir >= 21) || (bulan_lahir == 6 && tanggal_lahir <= 20)) {
            cout<<"zodiak = Gemini";
    }
    else if ((bulan_lahir == 6 && tanggal_lahir >= 21) || (bulan_lahir == 7 && tanggal_lahir <= 22)) {
            cout<<"zodiak = Cancer";
    }
    else if ((bulan_lahir == 7 && tanggal_lahir >= 23) || (bulan_lahir == 8 && tanggal_lahir <= 22)) {
            cout<<"zodiak = Leo";
    }
    else if ((bulan_lahir == 8 && tanggal_lahir >= 23) || (bulan_lahir == 9 && tanggal_lahir <= 22)) {
            cout<<"zodiak = Virgo";
    }
    else if ((bulan_lahir == 9 && tanggal_lahir >= 23) || (bulan_lahir == 10 && tanggal_lahir <= 22)) {
            cout<<"zodiak = Libra";
    }       
    else if ((bulan_lahir == 10 && tanggal_lahir >= 23) || (bulan_lahir == 11 && tanggal_lahir <= 21)){
            cout<<"zodiak = Scorpio";
    }       
    else {
            cout<< "zodiak = Sagitarius";  
        }
        
}
    return 0;
}